app/controllers/referral_promos_controller.rb

```diff
@@ -0,0 +1,43 @@
+class ReferralPromosController < ApplicationController
+  before_action :authenticate_publisher!
+
+  def show
+    @publisher = current_publisher
+    promo_running = active_promo_id.present?
+    if promo_running
+      if publisher_activated_promo
+        render(:active)
+      else
+        render(:activate)
+      end
+    else
+      render(:over)
+    end
+  end
+
+  def create
+    promo_running = active_promo_id.present?
```

Comment:
minor: i think a more railsy way to do this is to have a `before_action` filter which checks for stuff and redirects in case the condition fails.

Reply:
I thought about this too.

So apply a `before_action` to `create` that redirects to `index` if the promo is not running, and have `index` render `over` as it does already?

Reply:
i think we should render the `index` view always, since eventually that page will probably show their participation in past promos as well as opt into future promos.

for now, if the promo is over we can show a section which is blank except for the message.